src/h5web/providers/jupyter/jupyter-api.ts

```diff
@@ -161,4 +163,14 @@ export class JupyterStableApi extends ProviderApi {
 
     return contents;
   }
+
+  protected flattenValue(value: unknown, entity: Dataset, selection?: string) {
+    if (!hasArrayShape(entity)) {
+      return value;
+    }
+
+    const slicedDims = selection?.split(',').filter((s) => s.includes(':'));
```

Comment:
In Jupyter, the returned `value` dimensions change with the slicing. 

I detect when a dimension is conserved by checking for `:`.